In [1]:
from functions import data_functions as data

In [2]:
df = data.data('questionnaire_data.csv','t2')

In [3]:
df.shape

(149, 179)

In [4]:
null_values = df[df['7'].isnull()]
print(null_values.shape)
print(null_values.iloc[0:,0:9]) 

(2, 179)
                               1    2    3    4    5    6    7           8
176  514967-514958-85368090  NaN  NaN  NaN  NaN  NaN  NaN  NaN         NaN
181  514967-514958-87611797  Yes  Yes  Yes  Yes  NaN  Yes  NaN  08/02/2021


In [5]:
duplicates = df[df['8'].duplicated()]
print(duplicates.index)
print(duplicates['7'])

Int64Index([72, 121, 136, 138, 139, 141, 143, 144, 152, 156, 158, 159, 160,
            167, 178, 182],
           dtype='int64')
72     B2025
121    B1040
136    B2090
138    B2999
139    B2076
141    B2018
143     1009
144    B1031
152    B2033
156    B2043
158    B2039
159     2030
160    B2083
167    B2088
178    B2047
182     2065
Name: 7, dtype: object


B2025 - two B2025 values
B1040 - not a repeat someone else has same DoB
B2090 - two B2090 (138, 131 )
B2999 - drop as b2091 is first round
B2076 - two (111, 141)
B2018 two (86, 143 )
1009 - drop as orginal is B1009
B1031 - duplicate. Orginal one has a space after B1031.
B2033 - two B2033
B2043 - two B2043
B2039 - two B2039
2030 -  B2030 original
B2083 - two B2083
B2088 - two B2088
B2047 - two B2047
2065 - drop as B2065 is original

In [6]:
duplicates_b = df[df['7'].duplicated()]
print(duplicates_b.index)
print(duplicates_b['7'])

Int64Index([72, 136, 139, 141, 144, 152, 156, 158, 160, 167, 178, 181], dtype='int64')
72     B2025
136    B2090
139    B2076
141    B2018
144    B1031
152    B2033
156    B2043
158    B2039
160    B2083
167    B2088
178    B2047
181      NaN
Name: 7, dtype: object


In [7]:
dropindex = [72, 136, 138, 139, 141, 143, 144, 152, 156, 158, 159, 160, 176, 178, 181]

In [9]:
cleaned_df = data.data('questionnaire_data.csv','t2', clean=True, drop_index=dropindex)

In [10]:
cleaned_df.shape

(134, 179)

In [13]:
hc = cleaned_df[cleaned_df['7'].str.contains('B1')]
aan = cleaned_df[cleaned_df['7'].str.contains('B2')]